apps/sports-helsinki/src/domain/search/combinedSearch/hooks.ts

```diff
@@ -0,0 +1,163 @@
+import AppConfig from 'domain/app/AppConfig';
+import useUnifiedSearchVariables from 'domain/unifiedSearch/useUnifiedSearchVariables';
+import useEventSearchFilters from '../eventSearch/hooks/useEventSearchFilters';
+import { PARAM_SEARCH_TYPE } from './constants';
+import type { SearchTabId } from './searchTabs/tabsContext';
+import { isSearchTabId, useTabsContext } from './searchTabs/tabsContext';
+
+/**
+ * Use an URL parameter to scroll to
+ * the previously used search result card on a page change.
+ */
+export const useScrollToSearchResultItem = () => {
+  const router = useRouter();
+  const scrollTo = router.query?.scrollTo;
+  const listRef = useRef<HTMLUListElement | null>(null);
+
+  useEffect(() => {
+    const listElement = listRef.current;
+
+    if (scrollTo) {
+      const listItemElement = listElement?.querySelector(
+        decodeURIComponent(scrollTo.toString())
+      );
+
+      if (listItemElement) {
+        listItemElement.scrollIntoView({
+          block: 'center',
+        });
+      }
+    }
+  }, [scrollTo]);
+};
+
+export const useSearchTabsWithParams = (defaultTab: SearchTabId) => {
+  const router = useRouter();
+  const searchParams = new URLSearchParams(router.asPath.split('?')[1]);
+  const searchTypeParam = searchParams.get(PARAM_SEARCH_TYPE);
+  const initTab = searchTypeParam
+    ? (searchTypeParam as SearchTabId)
+    : defaultTab;
+
+  // If the search type param is not given in the URL, set it there.
+  React.useEffect(() => {
+    if (!isSearchTabId(searchTypeParam)) {
+      router.push(
+        { query: { ...router.query, [PARAM_SEARCH_TYPE]: defaultTab } },
+        undefined,
+        { shallow: true }
+      );
+    }
+  }, [searchTypeParam, router, defaultTab]);
+
+  return { initTab, searchTypeParam };
+};
+
+export function useLazyVenueSearchForTabCount() {
+  const { setResultCount } = useTabsContext();
+  const venueSearchFilters = useUnifiedSearchVariables();
+  const variables = React.useMemo(
+    () => ({
+      ...venueSearchFilters,
+      includeHaukiFields: AppConfig.isHaukiEnabled,
+    }),
+    [venueSearchFilters]
+  );
+  const [search, { loading, data, ...delegatedProps }] = useLazyQuery(
+    SearchListDocument,
+    {
+      variables,
+      // FIXME: Set the fetch policy to not trigger cache,
+      // or the Apollo client fails to request
+      // same query multiple times. Depending on settings,
+      // the client stops the query when the first response is given,
+      // and uses the cache for the next query with same name or
+      // returns the items twice.
+      // Since the query is exactly the same as the actual listing query,
+      // it would be really nice to have the result in cache already.
+      // See https://community.apollographql.com/t/executing-the-same-query-multiple-times-with-different-variables/3052
+      fetchPolicy: 'no-cache',
+      ssr: false,
+    }
+  );
+
+  React.useEffect(() => {
+    const count = () => {
```

Comment:
for some reason, it did not update, but it sure does look weird! I could try whether the problem still persists. Is it so that when the useEffect-callback runs, it runs only the function scope, not the whole parent function / hook? It would explain it. It would mean that that the count should be calculated inside the useEffect-callback.

Reply:
Maybe there is some room to refactor and investigate? The same useEffect is repeated 3 times, it could be a separate hook itself, like 

```
useSetTabSearchCount = ({loading, data, eventType})  => {
    const { setResultCount } = useTabsContext();
    React.useEffect(() => {
      ...
      setResultCount(eventType, ...);
   }, [loading, data, eventType]);
}

Reply:
I would still do this `useSetTabSearchCount` separately as the same thing is repeated in every tab's own hook.

Reply:
Perhaps could help to sonarcloud error too:
> 4.2% Duplicated Lines (%) on New Code (is greater than 3%)

Reply:
The data is structured differently, because there is a Venues-query and an Events-query. Your proposal would not look as simple in the real use.